### PR TITLE
feat: [CCM-7968]: Add Search in Perspective Grid

### PR DESCRIPTION
**Pull request by rajarshee-harness**
### Summary

This PR adds support for searching in Perspective Grid and some UX improvements. Also removes "None" from perspective filters.

#### Screenshots

| Before | After |
|:---:|:---:|
| Screenshot 2022-08-05 at 8 04 12 AM | Screenshot 2022-08-05 at 8 04 04 AM |
| Screenshot 2022-08-05 at 8 04 53 AM | Screenshot 2022-08-05 at 8 04 44 AM |

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 18 Issues 
* Code issue 18 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10411)

File: src/modules/75-ce/components/PerspectiveFilters/views/ProviderSelector.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ const ProviderSelector: React.FC<ProviderSelectorProps> = ({ fieldValueList, set
 
         {commonFields.map(commonField =>
           commonField.values.map(field => {
-            if (field) {
+            if (field && field.fieldId !== 'none') {
```
Comment (rishik30) [inline src/modules/75-ce/components/PerspectiveFilters/views/ProviderSelector.tsx:56]: do we have any enums for fieldId? if not, possible to create and use?

File: src/modules/75-ce/components/PerspectiveGrid/ColumnSelector.tsx [hunk 2]
```diff
@@ -37,20 +60,35 @@ const ColumSelector = (props: ColumnSelectorProps): JSX.Element => {
   }
 
   return (
-    <Layout.Horizontal style={{ alignItems: 'flex-end' }}>
-      {allowExportAsCSV ? (
+    <Layout.Horizontal style={{ alignItems: 'center' }} padding={{ left: 'xlarge', top: 'xlarge' }}>
+      {!isEqual(GROUP_BY_POD, groupBy) ? (
```
Comment (rishik30) [inline src/modules/75-ce/components/PerspectiveGrid/ColumnSelector.tsx:64]: why are we comparing the whole object here?

File: src/modules/75-ce/components/PerspectiveGrid/ColumnSelector.tsx [hunk 3]
```diff
@@ -37,20 +60,35 @@ const ColumSelector = (props: ColumnSelectorProps): JSX.Element => {
...
+        <Button
+          onClick={openDownloadCSVModal}
+          text={getString('ce.perspectives.exportCSV')}
+          rightIcon="launch"
+          iconProps={{ size: 12 }}
+          variation={ButtonVariation.LINK}
+        />
+      ) : null}
       <FlexExpander />
+      {setGridSearchParam ? (
```
Comment (rishik30) [inline src/modules/75-ce/components/PerspectiveGrid/ColumnSelector.tsx:84]: not an ideal check to render the component. It makes it difficult for other devs to understand the behavior. 
Try to add a flag for rendering purposes and the callback should be a separate prop.

File: src/modules/75-ce/components/PerspectiveGrid/PerspectiveGrid.tsx [hunk 4]
```diff
@@ -111,30 +105,40 @@ const PerspectiveGrid: React.FC<PerspectiveGridProps> = props => {
...
           openDownloadCSVModal={openDownloadCSVModal}
           columns={gridColumns}
           selectedColumns={selectedColumns}
           onChange={columns => setSelectedColumns(columns)}
+          setGridSearchParam={setGridSearchParam}
+        />
+      )}
+      {fetching ? (
+        <Container className={css.gridLoadingContainer}>
+          <Icon name="spinner" color="blue500" size={30} />
```
Comment (rishik30) [inline src/modules/75-ce/components/PerspectiveGrid/PerspectiveGrid.tsx:119]: Use `Color` from design-systems

File: src/modules/75-ce/pages/perspective-details/PerspectiveDetailsPage.tsx [hunk 5]
```diff
@@ -346,10 +347,24 @@ const PerspectiveDetailsPage: React.FC = () => {
...
+
+  const gridSearchFilter = useMemo(() => {
+    if (gridSearchParam) {
+      return {
+        idFilter: {
+          field: groupBy,
+          operator: QlceViewFilterOperator.Search,
+          values: [gridSearchParam]
+        }
+      } as QlceViewFilterWrapperInput
```
Comment (rishik30) [inline src/modules/75-ce/pages/perspective-details/PerspectiveDetailsPage.tsx:360]: return a default value too

**Review by rishik30 — CHANGES_REQUESTED** [review]

File: src/modules/75-ce/components/PerspectiveFilters/views/ProviderSelector.tsx (hunk 1 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/PerspectiveFilters/views/ProviderSelector.tsx]: We didn't. Have defined one.

File: src/modules/75-ce/components/PerspectiveGrid/ColumnSelector.tsx (hunk 2 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/PerspectiveGrid/ColumnSelector.tsx]: Replaced it with a better flag.

File: src/modules/75-ce/components/PerspectiveGrid/ColumnSelector.tsx (hunk 3 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/PerspectiveGrid/ColumnSelector.tsx]: Replaced it with a better flag.

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss [hunk 6]
```diff
@@ -27,24 +27,29 @@
...
   display: grid;
-  grid-template-columns: 76px auto 90px;
+  grid-template-columns: 90px auto;
+  column-gap: var(--spacing-medium);
   align-items: center;
   padding: 8px 0px;
 
   .groupByLabel {
-    margin-bottom: 2px;
+    font-weight: 500 !important;
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss:36]: use css vars

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss [hunk 7]
```diff
@@ -27,24 +27,29 @@
...
   }
 
   .groupBys {
     display: flex;
+    column-gap: var(--spacing-large);
 
     .groupByItems {
-      margin: 0px 10px;
+      font-weight: 500;
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss:44]: css vars

**Comment by rajarshee-harness** [comment]
retrigger jest

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
retrigger titlecheck

**Comment by rajarshee-harness** [comment]
retrigger titlecheck

**Comment by rajarshee-harness** [comment]
retrigger jest

**Review by rishik30 — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**